cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -523,6 +523,49 @@ __inline__ __device__ std::pair<char const*, char const*> trim_whitespaces_quote
   return {skip_character(trim_begin, quotechar), skip_character(trim_end, quotechar).base()};
 }
 
+/**
+ * @brief Adjusts the range to ignore starting/trailing whitespace characters.
+ *
+ * @param[in] begin Pointer to the first character in the parsing range
+ * @param[in] end Pointer to the first character after the parsing range
```

Comment:
If there are no output parameters, only mark `@param` instead of `@param[in]` and `@param[out]`.
```suggestion
 * @param begin Pointer to the first character in the parsing range
 * @param end Pointer to the first character after the parsing range
```